tests/test_modeltest.py

```diff
@@ -0,0 +1,49 @@
+import pytest
+from pytestqt.qt_compat import QStandardItemModel, QStandardItem, \
+    QFileSystemModel, QStringListModel, QSortFilterProxyModel, QT_API
+
+pytestmark = pytest.mark.usefixtures('qtbot')
+
+
+def test_standard_item_model(qtmodeltester):
+    """
+    Basic test which uses qtmodeltester with a QStandardItemModel.
+    """
+    model = QStandardItemModel()
+    items = [QStandardItem(str(i)) for i in range(5)]
+    items[0].setChild(0, items[4])
+    model.setItem(0, 0, items[0])
+    model.setItem(0, 1, items[1])
+    model.setItem(1, 0, items[2])
+    model.setItem(1, 1, items[3])
+
+    qtmodeltester.check(model)
+
+
+def test_file_system_model(qtmodeltester, tmpdir):
+    tmpdir.ensure('directory', dir=1)
+    tmpdir.ensure('file1.txt')
+    tmpdir.ensure('file2.py')
+    model = QFileSystemModel()
+    model.setRootPath(str(tmpdir))
+    qtmodeltester.check(model)
+    tmpdir.ensure('file3.py')
+    qtmodeltester.check(model)
+
+
+@pytest.mark.skipif(QT_API == 'pyside', reason='For some reason this fails in '
+                                               'PySide with a message about'
+                                               'columnCount being private')
```

Comment:
Actually making `columnCount` private would break sub-classing contracts, as functions which receive a `QAbstractItemModel` would expect to be able to call `columnCount`, even if an instance of a `QAbstractListModel` is given.

I just noticed this in the [docs](http://doc.qt.io/qt-5/qabstractlistmodel.html) too:

> The columnCount() function is implemented for interoperability with all kinds of views, but by default informs views that the model contains only one column.

Also PySide's docs [explicitly say the same thing](https://srinikom.github.io/pyside-docs/PySide/QtCore/QAbstractListModel.html#qabstractlistmodel), so to me it seems like a bug in `PySide`. :sweat: 

I created an issue at PySide/PySide#127, let's see what they have to say.

I also tried to implement the workaround you suggested, but then I hit the same issue with `parent()` and `hasChildren()` methods, which at this point doesn't seem worthwhile to implement this workaround...